packages/browser/src/request.ts

```diff
@@ -5,10 +5,28 @@ import { formDataToQuery } from './utils/request-utils'
 
 import { logger } from './utils/logger'
 import { AbortController, fetch, navigator, XMLHttpRequest } from './utils/globals'
-import { gzipSync, strToU8 } from 'fflate'
 
 import { _base64Encode } from './utils/encode-utils'
 
+/* eslint-disable compat/compat */
+const supportsCompressionStream = typeof CompressionStream !== 'undefined'
+
+let sendChain: Promise<void> = Promise.resolve()
+
+export const _resetSendChain = (): void => {
+    sendChain = Promise.resolve()
+}
+
+const gzipCompress = async (data: string): Promise<Blob | null> => {
+    if (!supportsCompressionStream) {
+        return null
+    }
+    const stream = new Blob([data]).stream()
+    const compressedStream = stream.pipeThrough(new CompressionStream('gzip'))
+    return new Response(compressedStream).blob()
```

Comment:
```suggestion
    return new Response(compressedStream).blob().then(
    blob => new Blob([blob], { type: 'application/gzip' })
)
```

we can set the blob mime type here as well